apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,205 @@
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=core
+	Group string `json:"group" protobuf:"bytes,1,opt,name=group"`
+
+	// Resource is the API resource name of the referent. Omitting the value
+	// or specifying the empty string indicates the services resource. For example,
+	// use the following to specify a services resource:
+	//
+	// fooRef:
+	//   name: myservice
+	//
+	// Otherwise, if the services resource is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=services
+	Resource string `json:"resource" protobuf:"bytes,2,opt,name=resource"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
+
+	// Port is the port of the referent. If unspecified, this policy applies to
+	// all ports on the backend.
+	// +optional
+	Port *int32 `json:"port,omitempty" protobuf:"varint,4,opt,name=port"`
+}
+
+// BackendTLSConfig describes TLS configuration for a backend.
+type BackendTLSConfig struct {
+	// CertificateRef is a reference to a TLS client certificate that may be
+	// used to connect to these backends. If an entry in this list omits or
+	// specifies the empty string for both the group and the resource, the
+	// resource defaults to "secrets". An implementation may support other
+	// resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// If a Secret is referenced, it must be of type "kubernetes.io/tls" and
+	// contain tls.crt and tls.key data fields that contain the certificate and
+	// private key to use for TLS. It may optionally include a certificate
+	// authority bundle under a ca.crt data field.
+	//
+	// Support: Extended
+	//
+	// +optional
+	CertificateRef *CertificateObjectReference `json:"clientCertificateRef,omitempty" protobuf:"bytes,1,rep,name=clientCertificateRef"`
+
+	// Options are a list of key/value pairs to give extended options to the
+	// provider.
+	//
+	// Support: Implementation-specific.
+	// +optional
+	Options map[string]string `json:"options,omitempty" protobuf:"bytes,3,rep,name=options"`
+}
+
+// BackendPolicyStatus defines the observed state of BackendPolicy.
+type BackendPolicyStatus struct {
+	// Conditions describe the current conditions of the BackendPolicy.
+	// +optional
+	Conditions []BackendPolicyCondition `json:"conditions,omitempty" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+// BackendPolicyConditionType is a type of condition associated with a
+// BackendPolicy.
+type BackendPolicyConditionType string
+
+const (
+	// ConditionNoSuchBackend indicates that one or more of the the specified
+	// Backends does not exist.
+	ConditionNoSuchBackend BackendPolicyConditionType = "NoSuchBackend"
+)
+
+// BackendPolicyCondition is an error status for a given BackendPolicy.
+type BackendPolicyCondition struct {
```

Comment:
With #302 under way, can we update this to use metav1.Condition type?

Reply:
Updated, thanks!